backend/src/api/api_v1/endpoints/admin.py

```diff
@@ -0,0 +1,72 @@
+from typing import Any
+
+from fastapi import APIRouter, Depends, HTTPException
+from sqlalchemy.ext.asyncio import AsyncSession
+from src import crud
+from src.api import deps
+from src.core.config import settings
+from src.db.db import get_session
+from src.models.user import User, UserCreate, UserResponse, UserUpdate
+from src.utils import send_new_account_email
+
+router = APIRouter()
+
+@router.delete("/{user_id}", response_model=UserResponse)
+async def remove_user(
+    user_id: int,
+    db: AsyncSession = Depends(get_session),
+    current_user: User = Depends(deps.get_current_active_admin),
+) -> Any:
+    """
+    Remove an existing user. Must be logged in first. Must be an admin.
+    """
+    user = await crud.user.get(db, id=user_id)
+
+    if user is None:
+        raise HTTPException(status_code=404, detail="That user does not exist.")
+
+    if user.user_id != current_user.id:
+        raise HTTPException(status_code=401, detail="You are unauthorized to remove this budget")
+
+    if user.is_admin:
+        raise HTTPException(status_code=403, detail="Admin cannot be removed")
+    user = await crud.user.remove(db, id=user_id)
+
+    return user
+
+@router.post("/{user_id}", response_model=UserCreate)
+async def create_user(
+    user_create: UserCreate,
+    db: AsyncSession = Depends(get_session),
+    current_user: User = Depends(deps.get_current_active_admin),
```

Comment:
add a query param to tell the route whether an admin should be created or not, here is the fastapi docs on how that works https://fastapi.tiangolo.com/tutorial/query-params/?h=query